src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -395,9 +424,74 @@ ValidatorList::removePublisherList (PublicKey const& publisherKey)
             --iVal->second;
     }
 
+    iList->second.list.clear();
+    iList->second.available = false;
+
     return true;
 }
 
+boost::optional<TimeKeeper::time_point>
+ValidatorList::expires() const
+{
+    boost::shared_lock<boost::shared_mutex> read_lock{mutex_};
+    boost::optional<TimeKeeper::time_point> res{boost::none};
+    for (auto const& p : publisherLists_)
+    {
+        if (p.second.available &&
+            TimeKeeper::time_point{} < p.second.expiration)
+        {
+            if (!res || p.second.expiration < *res)
+                res = p.second.expiration;
+        }
+    }
+    return res;
+}
+
+Json::Value
+ValidatorList::getJson() const
+{
+    Json::Value res(Json::objectValue);
+
+    boost::shared_lock<boost::shared_mutex> read_lock{mutex_};
+
+    res[jss::validation_quorum] = static_cast<Json::UInt>(quorum());
+
+    if (auto when = expires())
+        res[jss::validator_list_expires] = to_string(*when);
+    else
+        res[jss::validator_list_expires] = "unknown";
+
+    // Publisher lists
+    Json::Value& jPublisherLists =
+        (res[jss::publisher_lists] = Json::arrayValue);
+    for (auto const& p : publisherLists_)
+    {
+        Json::Value& curr = jPublisherLists.append(Json::objectValue);
+        curr[jss::pubkey_publisher] = strHex(p.first);
+        curr[jss::seq] = static_cast<Json::UInt>(p.second.sequence);
+        curr[jss::available] = p.second.available;
+        curr[jss::expiration] = to_string(p.second.expiration);
+        Json::Value& keys = (curr[jss::list] = Json::arrayValue);
+        for (auto const& key : p.second.list)
+        {
+            keys.append(toBase58(TokenType::TOKEN_NODE_PUBLIC, key));
+        }
+    }
+
+    // Current validator keys
```

Comment:
This seems redundant since it's always accompanied by the individual published lists from above.
What if we instead just list which keys are trusted (from `trustedKeys_`)?

Reply:
Agreed, that makes a lot more sense.

Reply:
Call the JSON field `validator_keys` or `trusted_validator_keys`?

Reply:
I like `trusted_validator_keys` better